cmd/apprepository-controller/main.go

```diff
@@ -91,6 +92,7 @@ func init() {
 	flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
 	flag.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
 	flag.StringVar(&repoSyncImage, "repo-sync-image", "quay.io/helmpack/chart-repo:latest", "container repo/image to use in CronJobs")
+	flag.StringVar(&repoSyncImagePullSecrets, "repo-sync-image-pullsecrets", "", "optional reference to secrets in the same namespace to use for pulling the image used by this pod")
```

Comment:
I did try this approach and I was able to have a []string with the values. However, when generating automatically these args from the Chart template, I considered that a comma-separated string would be easier to understand.  I personally see having multiple values for the same flag it's a bit confusing. 

Nevertheless, as you mention, with the array flag approach we could have both options and will let users decide. So, ok, let's implement it with an array flag

Reply:
I believe the `,` approach won't work out of the box with the `flag` package. You will need to code it yourself, following the example in the URL, it would be something like:

```
func (i *arrayFlags) Set(value string) error {
    *i = append(*i, strings.Split(value, ",")...)
    return nil
}
```

But you would need also being careful not to include spaces and that kind of things. For simplicity I would just go to use as many `--repo-syn-image-pullsecrets` as needed but up to you (also note that setting these flags is something internal and automatic, not something the user needs to do on their own).